site/blocks/StickyNav.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { createRef } from "react"
+import ReactDOM from "react-dom"
+import classnames from "classnames"
+import { debounce, get } from "../../clientUtils/Util.js"
+
+function getTotalOffset(element: HTMLElement): {
+    x: number
+    y: number
+} {
+    let y = 0
+    let x = 0
+    while (true) {
+        x += element.offsetLeft
+        y += element.offsetTop
+        if (!element.offsetParent) {
+            break
+        }
+        element = element.offsetParent as HTMLElement
+    }
+    return { x, y }
+}
+
+const STICKY_NAV_HEIGHT = 54
+const HEADING_SCROLL_MARGIN = 16
+
+interface StickyNavProps {
+    links: { text: string; target: string }[]
+}
+class StickyNav extends React.Component<
+    StickyNavProps,
+    {
+        headingPositions: { id: string; position: number }[]
+        currentHeadingIndex?: number
+    }
+> {
+    ulRef = createRef<HTMLUListElement>()
+
+    state = {
+        currentHeadingIndex: undefined,
+        headingPositions: [] as { id: string; position: number }[],
+    }
+
+    get links() {
+        return this.props.links
+    }
+
+    handleResize = debounce(() => {
+        this.setHeadingPositions()
+    }, 100)
+
+    handleScroll = debounce(() => {
+        this.setCurrentHeading()
+    }, 100)
+
+    setCurrentHeading() {
+        const { scrollY } = window
+        let currentHeadingIndex = 0
+        for (let i = 0; i < this.state.headingPositions.length; i++) {
+            const heading = this.state.headingPositions[i]
+            const target =
+                heading.position -
+                STICKY_NAV_HEIGHT -
+                HEADING_SCROLL_MARGIN -
+                // Additional clearance for height of actual element
+                32
+            if (scrollY <= target) {
+                break
+            } else {
+                currentHeadingIndex = i
+            }
+        }
+
+        const current = this.ulRef.current
+        if (current) {
+            const parent = current.parentElement
+            if (parent) {
+                const hasScrollbar = parent.scrollWidth < current.scrollWidth
+                if (hasScrollbar) {
+                    const listItem = current.children[
+                        currentHeadingIndex
+                    ] as HTMLElement
+                    const containerPadding = 32
+                    current.scrollTo({
+                        behavior: "smooth",
+                        left: listItem.offsetLeft - containerPadding,
+                    })
+                }
+            }
+        }
+
+        this.setState({
+            currentHeadingIndex,
+        })
+    }
+
+    setHeadingPositions() {
+        const headingPositions: any[] = []
+        this.links.forEach((link) => {
+            const element = document.querySelector<HTMLElement>(link.target)
+            if (element) {
+                const { y } = getTotalOffset(element)
+                headingPositions.push({ id: link.target, position: y })
```

Comment:
Maybe it could be 
```suggestion
                headingPositions.push({ id: link.target, yPosition: y })
```
to make it completely clear that this is only the y coordinate.